File: SEP3_Client/Model/Unit/User/Friend.cs
using System;

namespace SEP3_Client.Model.Unit.User
{
    public class Friend : BasicInformation
    {
        public Friend(Account account) : base(account.GetId(),account.GetUserName())
        {
            SetBirthday(account.GetBirthday());
        }

        public Friend(string id, string userName, Birthday birthday) : base(id, userName)
        {
            SetBirthday(birthday);
        }

        private Friend(Friend friend) : base(friend) {}
        
        public Friend() {}

        public void Update(Friend friend) 
        {
            Update(friend.GetUserName(),friend.GetBirthday(),friend.Gender,friend.Hometown,friend.PhoneNumber);
        }
        
        public Friend Copy() 
        {
            return new Friend(this);
        }
    }
}

File: SEP3_Database/src/main/java/com/example/SEP3_Database/model/domain/unit/user/Friend.java
package com.example.SEP3_Database.model.domain.unit.user;

public class Friend extends BasicInformation {

    public Friend(Account account) {
        super(account.getId(), account.getUserName());
        setBirthday(account.getBirthday());
    }

    public Friend(String id, String userName, Birthday birthday) {
        super(id, userName);
        setBirthday(birthday);
    }

    public void update(Friend friend) {
        update(friend.getUserName(),friend.getBirthday(),friend.getGender(),friend.getHometown(),friend.getPhoneNumber());
    }

    public Friend copy() {
        return new Friend(getId(),getUserName(),getBirthday());
    }
}
